test/parallel/test-worker-prof.js

```diff
@@ -1,40 +1,72 @@
 'use strict';
-require('../common');
+const common = require('../common');
 const tmpdir = require('../common/tmpdir');
 const fs = require('fs');
 const assert = require('assert');
 const { join } = require('path');
 const { spawnSync } = require('child_process');
-const { Worker } = require('worker_threads');
 
 // Test that --prof also tracks Worker threads.
 // Refs: https://github.com/nodejs/node/issues/24016
 
 if (process.argv[2] === 'child') {
-  const spin = `
-  const start = Date.now();
-  while (Date.now() - start < 1000);
+  let files = fs.readdirSync(tmpdir.path);
+  const plog = files.filter((name) => /\.log$/.test(name))[0];
+  if (plog === undefined) {
+    console.error('`--prof` did not produce a profile log for parent thread!');
+    process.exit(1);
+  }
+  const pingpong = `
+  let counter = 0;
+  const { Worker, parentPort  } = require('worker_threads');
+  parentPort.on('message', (m) => {
+    if (counter++ === 1024)
+      process.exit(0);
+     parentPort.postMessage(
+       m.toString().split('').reverse().toString().replace(/,/g, ''));
+  });
   `;
-  new Worker(spin, { eval: true });
-  eval(spin);
-  return;
-}
 
-tmpdir.refresh();
-spawnSync(process.execPath, ['--prof', __filename, 'child'],
-          { cwd: tmpdir.path });
-const files = fs.readdirSync(tmpdir.path);
-const logfiles = files.filter((name) => /\.log$/.test(name));
-assert.strictEqual(logfiles.length, 2);  // Parent thread + child thread.
+  const { Worker } = require('worker_threads');
+  const data = 'x'.repeat(1024);
+  const w = new Worker(pingpong, { eval: true });
+  w.on('message', (m) => {
+    w.postMessage(m.toString().split('').reverse().toString().replace(/,/g, ''));
+  });
+
+  w.on('exit', common.mustCall(() => {
+    files = fs.readdirSync(tmpdir.path);
+    const wlog = files.filter((name) => /\.log$/.test(name) && name !== plog)[0];
+    if (wlog === undefined) {
+      console.error('`--prof` did not produce a profile log' +
+                    ' for worker thread!');
+      process.exit(1);
+    }
+    process.exit(0);
+  }));
+  w.postMessage(data);
+} else {
+  tmpdir.refresh();
+  const spawnResult = spawnSync(
+    process.execPath, ['--prof', __filename, 'child'],
+    { cwd: tmpdir.path });
+  assert.strictEqual(spawnResult.stderr.toString(), '');
+  assert.strictEqual(spawnResult.status, 0);
+  assert.strictEqual(spawnResult.signal, null);
```

Comment:
Suggest swapping this line and the one above it so that the signal is checked first as that would be more useful to know if it is not null rather than the null exit code seen in https://github.com/nodejs/node/issues/26401#issuecomment-473248681.